src/future.ts

```diff
@@ -8,6 +8,15 @@ import { Stream } from "./stream";
 
 export type MapFutureTuple<A> = { [K in keyof A]: Future<A[K]> };
 
+export type SemanticFuture<A> =
+  | { time: number; value: A }
+  | { time: 9007199254740991; value: never };
+
+const neverOccuringSemanticFuture = {
+  time: 9007199254740991,
```

Comment:
What about using `Infinity` instead of `9007199254740991`?

Reply:
We could use `Infinity` in `neverOccuringSemanticFuture`, but `Infinity` is not a type, that is why we decided on this super large number which is the same as `Number.MAX_SAFE_INTEGER`.

https://github.com/Microsoft/TypeScript/issues/15135
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/MAX_SAFE_INTEGER